dawn/src/dawn/SIR/SIR.cpp

```diff
@@ -565,9 +565,20 @@ CompareResult Field::comparison(const Field& rhs) const {
   return StencilFunctionArg::comparison(rhs);
 }
 
+bool UnstructuredFieldDimension::chainIsValid() const {
```

Comment:
I'm generally happy to have these checks around. So far our assumption was garbage in garbage out in SIR. Do we want a validation module moving forward similar to the verifier we have in IIR.

If so we might want to start with this functionality

Reply:
I was thinking of this more of as a contract to the argument of the constructor. There is also another check that the chain has at least one entry in the constructor. I can see an argument that we should be as permissive as possible w.r.t the SIR, and handle all the validation in appropriate passes. However, if we should decide to go down that route I'd say we move this to a validator pass in a future PR.

Reply:
sounds good. Just wanted to put it on the radar for discussion